epochStart/metachain/economics.go

```diff
@@ -329,12 +346,91 @@ func (e *economics) adjustRewardsPerBlockWithLeaderPercentage(
+	return e.rewardsHandler.MaxInflationRate(yearsIndex), nil
+}
+
+func (e *economics) computeInflationRate(currentRound uint64, currentEpoch uint32) (float64, error) {
+	supernovaInEpochActivated := e.enableEpochsHandler.IsFlagEnabledInEpoch(common.SupernovaFlag, currentEpoch)
+	if !supernovaInEpochActivated {
+		return e.computeInflationRateLegacy(currentRound, currentEpoch)
+	}
+
+	currentYear := currentEpoch/uint32(numberOfDaysInYear) + 1
+	supernovaActivationYear, err := e.getSupernovaActivationYear(currentEpoch)
+	if err != nil {
+		return 0, err
+	}
+
+	if currentYear == supernovaActivationYear {
+		return e.computeInflationRateInSupernovaYear(currentYear)
+	}
+
+	return e.computeInflationRateAfterSupernovaYear(currentRound, currentEpoch, supernovaActivationYear)
+}
+
+func (e *economics) getPrevSupernovaActivationEpoch(supernovaActivationEpoch uint32) uint32 {
+	if supernovaActivationEpoch == 0 {
+		return supernovaActivationEpoch
+	}
+
+	return supernovaActivationEpoch - 1
+}
+
+func (e *economics) getSupernovaActivationYear(currentEpoch uint32) (uint32, error) {
+	supernovaActivationRound := e.enableRoundsHandler.GetActivationRound(common.SupernovaRoundFlag)
+	supernovaActivationEpoch := e.enableEpochsHandler.GetActivationEpoch(common.SupernovaFlag)
+
+	supernovaPrevActivationEpoch := e.getPrevSupernovaActivationEpoch(supernovaActivationEpoch)
+	chainParametersUntilSupernova, err := e.chainParamsHandler.ChainParametersForEpoch(supernovaPrevActivationEpoch)
+	if err != nil {
+		return 0, err
+	}
+	roundDuration := chainParametersUntilSupernova.RoundDuration
+	if roundDuration <= 0 {
+		return 0, errors.ErrInvalidRoundDuration
+	}
+
+	roundsPerDay := common.ComputeRoundsPerDay(time.Duration(roundDuration)*time.Millisecond, e.enableEpochsHandler, supernovaPrevActivationEpoch)
+	roundsPerYear := numberOfDaysInYear * roundsPerDay
+
+	supernovaActivationYear := uint32(supernovaActivationRound/roundsPerYear) + 1
+
+	return supernovaActivationYear, nil
+}
+
+func (e *economics) computeInflationRateInSupernovaYear(currentYear uint32) (float64, error) {
+	log.Trace("computeInflationRateInSupernovaYear",
+		"currentYear", currentYear,
+	)
+	return e.rewardsHandler.MaxInflationRate(currentYear), nil
+}
+
+func (e *economics) computeInflationRateAfterSupernovaYear(
+	currentRound uint64,
+	currentEpoch uint32,
+	supernovaActivationYear uint32,
+) (float64, error) {
+	log.Trace("computeInflationRateAfterSupernovaYear",
+		"currentEpoch", currentEpoch,
+		"currentRound", currentRound,
+		"supernovaActivationYear", supernovaActivationYear,
+	)
+
+	supernovaActivationRound := e.enableRoundsHandler.GetActivationRound(common.SupernovaRoundFlag)
+
+	roundsPerDay := common.ComputeRoundsPerDay(e.roundTime.TimeDuration(), e.enableEpochsHandler, currentEpoch)
+
+	roundsPerYear := numberOfDaysInYear * roundsPerDay
+
+	yearsIndex := uint32((currentRound-supernovaActivationRound)/roundsPerYear) + supernovaActivationYear
```

Comment:
I don't think this computation is correct, as you are setting the supernova activation year as the start of an inflation year index, but the year index needs to be related to the genesis.

e.g 
```
Example1:
let's say we have:
100 rounds per year both before and after Supernova
Supernova activation round is 230 => Supernova Activation year is 3
current round is 301 (should compute as year 4)
-> yearIndex  := (301-230)/100+3 =  71/100 + 3 = 3 // which is wrong

Example2:
100 rounds per year before Supernova
200 rounds per year after Supernova
Supernova activation round is 230 => Supernova year is 3
current round is 230+70*2 +1 = 371 ( chosen to be in fact the second day in the next year, year 4)
-> yearIndex := (371 - 230)/200 + 3 = 141/200 + 3 = 3 // this is also wrong
```